StepperEngine/src/Flow/Flow.java

```diff
@@ -309,13 +313,19 @@ public void execute(){
         createFlowLog();
     }
 
-    private void setFlowStatus(Step step){
-        if(step.getStatus()== Step.Status.Success)
-            status=Status.SUCCESS;
-        else if(step.getStatus()== Step.Status.Warning)
-            status=Status.WARNING;
-        else
+    private void setFlowStatus(){
+        if(steps.get(steps.size()-1).getStatus()== Step.Status.Failure){
```

Comment:
If step number x fails (and it's **blocking**), but there are x+1 steps? 

step number x+1 will not have fail status, no? (because it will not even be executed)

and then the "if" will not happen

Reply:
if step number x fails and is blocking, an exception will be thrown and setFlowStatus will not be called, instead in the catch block the status will be set to failure.
if(steps.get(steps.size()-1).getStatus()== Step.Status.Failure), is here in case the last step failed and was non blocking
(i didn't see it as an user input error that needed to be validated)